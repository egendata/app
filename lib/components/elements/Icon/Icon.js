import React from 'react'
import { Image, View } from 'react-native'
import SvgIcon from 'react-native-svg-icon'
import styled, { theme } from '../../../theme'
import icons from './img/icons'

const StyledQRIcon = styled(View)`
  width: 130px;
  height: 130px;
`

const HomeIcon = ({ active }) => (
  <Image
    style={{ width: 30, height: 24 }}
    source={
      active
        ? require('./img/home_active.png')
        : require('./img/home_inactive.png')
    }
  />
)

const ConsentIcon = () => (
  <StyledQRIcon>
    <Icon
      name={'qr'}
      stroke={theme.colors.quiet}
      width="100%"
      height="100%"
      fill="none"
    />
  </StyledQRIcon>
)

const AccountIcon = ({ active }) => (
  <Image
    style={{ width: 24, height: 24 }}
    source={
      active
        ? require('./img/account_active.png')
        : require('./img/account_inactive.png')
    }
  />
)

const getNavbarIcon = (name, active) => {
  switch (name) {
    case 'konto':
      return <AccountIcon active={active} />
    case 'qr-kod':
      return <ConsentIcon />
    case 'hem':
      return <HomeIcon active={active} />
  }
}

export const NavbarIcon = ({ name, active }) => getNavbarIcon(name, active)

export const Icon = props => <SvgIcon {...props} svgs={icons} />
