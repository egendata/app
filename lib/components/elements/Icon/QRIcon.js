import React from 'react'
import { View } from 'react-native'
import styled from '../../../theme'
import LogotypeClean from '../logotypeClean'

const StyledQRIcon = styled(View)`
  align-items: center;
  width: 96px;
  height: 96px;
  margin-bottom: 60px;
  border: 5px solid ${({ theme }) => theme.colors.quiet};
  justify-content: center;
  border-radius: 100px;
  background-color: ${({ theme }) => theme.colors.primary};
`

const QRIcon = () => (
  <StyledQRIcon>
    <LogotypeClean />
  </StyledQRIcon>
)

export default QRIcon
