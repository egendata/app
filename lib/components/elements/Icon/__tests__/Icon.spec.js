import React from 'react'
import { shallow } from 'react-native-testing-library'
import MockedProvider from '../../../../utils/test-utils'
import { NavbarIcon } from '../Icon'

describe('elements/Icons', () => {
  let iconComponent

  beforeEach(() => {
    /* eslint-disable react/display-name */
    iconComponent = name => {
      return (
        <MockedProvider>
          <NavbarIcon name={name} />
        </MockedProvider>
      )
    }
  })

  it('should return appropriate icon based on name = "konto"', () => {
    const { output } = shallow(iconComponent('konto'))

    expect(output).toMatchSnapshot()
  })

  it('should return appropriate icon based on name = "qr-kod"', () => {
    const { output } = shallow(iconComponent('qr-kod'))

    expect(output).toMatchSnapshot()
  })

  it('should return appropriate icon based on name = "hem"', () => {
    const { output } = shallow(iconComponent('hem'))

    expect(output).toMatchSnapshot()
  })
})
