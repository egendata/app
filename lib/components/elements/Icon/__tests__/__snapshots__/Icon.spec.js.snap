// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/Icons should return appropriate icon based on name = "hem" 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "accept": "#0AC238",
        "border": "#EBEAFA",
        "deny": "#FF5374",
        "icon": Object {
          "arrowRight": "#81A1B1",
        },
        "inactive": "#E0E7EB",
        "lightGrey": "#EAEAF1",
        "loading": "#81A1B1",
        "primary": "#0099E5",
        "quiet": "#F7F8F8",
        "shadow": "grey",
        "text": Object {
          "pre": "#81A1B1",
          "primary": "#454F54",
        },
        "white": "#FFFFFF",
      },
      "fontSize": Object {
        "large": "48px",
        "medium": "32px",
        "regular": "18px",
        "regularMedium": "24px",
        "small": "14px",
      },
      "sizes": Object {
        "borderRadius": Object {
          "small": 5,
        },
        "elevation": 6,
      },
      "spacing": Object {
        "large": "48px",
        "medium": "24px",
        "small": "12px",
      },
      "typography": Object {
        "bold": "Karla-Bold",
        "contrast": "Rubik-Medium",
        "regular": "Karla-Regular",
      },
    }
  }
>
  <NavbarIcon
    name="hem"
  />
</ThemeProvider>
`;

exports[`elements/Icons should return appropriate icon based on name = "konto" 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "accept": "#0AC238",
        "border": "#EBEAFA",
        "deny": "#FF5374",
        "icon": Object {
          "arrowRight": "#81A1B1",
        },
        "inactive": "#E0E7EB",
        "lightGrey": "#EAEAF1",
        "loading": "#81A1B1",
        "primary": "#0099E5",
        "quiet": "#F7F8F8",
        "shadow": "grey",
        "text": Object {
          "pre": "#81A1B1",
          "primary": "#454F54",
        },
        "white": "#FFFFFF",
      },
      "fontSize": Object {
        "large": "48px",
        "medium": "32px",
        "regular": "18px",
        "regularMedium": "24px",
        "small": "14px",
      },
      "sizes": Object {
        "borderRadius": Object {
          "small": 5,
        },
        "elevation": 6,
      },
      "spacing": Object {
        "large": "48px",
        "medium": "24px",
        "small": "12px",
      },
      "typography": Object {
        "bold": "Karla-Bold",
        "contrast": "Rubik-Medium",
        "regular": "Karla-Regular",
      },
    }
  }
>
  <NavbarIcon
    name="konto"
  />
</ThemeProvider>
`;

exports[`elements/Icons should return appropriate icon based on name = "qr-kod" 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "accept": "#0AC238",
        "border": "#EBEAFA",
        "deny": "#FF5374",
        "icon": Object {
          "arrowRight": "#81A1B1",
        },
        "inactive": "#E0E7EB",
        "lightGrey": "#EAEAF1",
        "loading": "#81A1B1",
        "primary": "#0099E5",
        "quiet": "#F7F8F8",
        "shadow": "grey",
        "text": Object {
          "pre": "#81A1B1",
          "primary": "#454F54",
        },
        "white": "#FFFFFF",
      },
      "fontSize": Object {
        "large": "48px",
        "medium": "32px",
        "regular": "18px",
        "regularMedium": "24px",
        "small": "14px",
      },
      "sizes": Object {
        "borderRadius": Object {
          "small": 5,
        },
        "elevation": 6,
      },
      "spacing": Object {
        "large": "48px",
        "medium": "24px",
        "small": "12px",
      },
      "typography": Object {
        "bold": "Karla-Bold",
        "contrast": "Rubik-Medium",
        "regular": "Karla-Regular",
      },
    }
  }
>
  <NavbarIcon
    name="qr-kod"
  />
</ThemeProvider>
`;
